docs/antrea-network-policy.md

```diff
@@ -715,6 +716,86 @@ The following kubectl commands can be used to retrieve CG resources:
     kubectl get cg.core.antrea.tanzu.vmware.com
 ```
 
+## Select Namespace by Name
+
+Kubernetes NetworkPolicies and Antrea-native policies allow selecting
+workloads from Namespaces with the use of a label selector (i.e. `namespaceSelector`).
+However, it is often desirable to be able to select Namespaces directly by their `name`
+as opposed to using the `labels` associated with the Namespaces. In order to select
+Namespaces by name, Antrea labels Namespaces with a reserved label `antrea.io/metadata.name`,
+whose value is set to the Namespace's name. Users can then use this label in the
+`namespaceSelector` field, in both K8s NetworkPolicies and Antrea-native policies to
+select Namespaces by name. By default, Namespaces are not labeled with the reserved name label.
+In order for the Antrea controller to label the Namespaces, the `labelsmutator.antrea.io`
+`MutatingWebhookConfiguration` must be enabled. This can be done by applying the following
+webhook configuration YAML:
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1
+kind: MutatingWebhookConfiguration
+metadata:
+  # Do not edit this name.
+  name: "labelsmutator.antrea.io"
+webhooks:
+  - name: "namelabelmutator.antrea.io"
+    clientConfig:
+      service:
+        name: "antrea"
+        namespace: "kube-system"
+        path: "/mutate/addlabels"
+    rules:
+      - operations: ["CREATE", "UPDATE"]
+        apiGroups: [""]
+        apiVersions: ["v1"]
+        resources: ["namespaces"]
+        scope: "Cluster"
+    admissionReviewVersions: ["v1", "v1beta1"]
+    sideEffects: None
+    timeoutSeconds: 5
+```
+
+**Note**: `antrea-controller` Pod must be restarted after applying this YAML.
```

Comment:
@antoninbas @jianjuns @tnqn added this note, i.e. the need to restart antrea-controller Pod once yaml is applied, for controller to patch the webhoook with cert.
Alternatively, we could create the webhook from Antrea on enabling a config option, and remove the gen manifest script option. Let me know if
1. Current approach is fine and merge in 0.13
2. Current approach is fine temporarily and update to second or other approach in 0.14
2. Should update the approach and instead defer to 0.14?

Reply:
even though the approach sounds fine to me, I do suggest we wait for 0.14 given that this is a last minute change and not a must-have for 0.13 IMO

Reply:
> Alternatively, we could create the webhook from Antrea on enabling a config option, and remove the gen manifest script option.
This sounds like a better approach to me. Do you agree too? If so probably let us move this one to 0.14.

Reply:
I agree that this a good approach. Maybe let's wait for Quan to give his opinion on this when he is back from his holiday.